src/main/java/de/tum/in/www1/artemis/service/competency/CompetencyService.java

```diff
@@ -552,4 +555,17 @@ public boolean vertexIsPartOfCycle(Vertex sourceVertex) {
         }
         return graph.hasCycle();
     }
+
+    /**
+     * Checks if the competency is part of the course
+     *
+     * @param competencyId The id of the competency
+     * @param courseId     The id of the course
+     * @throws BadRequestAlertException if the competency does not belong to the course
+     */
+    public void checkIfCompetencyBelongsToCourse(long competencyId, long courseId) {
```

Comment:
This method seems like it would fit better into a Ressouce class since it verifies the parameter from the user.